src/mimeparser.rs

```diff
@@ -570,13 +570,13 @@ impl MimeMessage {
 
                             any_part_added = true;
                         }
-                        Some("delivery-status") => {
+                        // Some providers, e.g. Tiscali, forget to set the report-type. So, if it's None, assume that it might be delivery-status
+                        Some("delivery-status") | None => {
```

Comment:
I changed the behavior for non-ndns here (for a cleaner code): Previously parts with "multipart/report" but without a `report-type` were ignored. Now all subparts are parsed recursively and `any_part_added` might be set to true so that other parts are not parsed.

OTOH, I do not think there are any mails where this might be a problem.